code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -86,6 +86,10 @@
 	if(!damage) //If we've actually taken damage, check whether we alert the hive
 		return
 
+	if(isobj(pulling))
+		stop_pulling()
+
```

Comment:
```suggestion
	// additional chance when pulling to stop pulling the mob.
	if(isobj(pulling) && prob(5))
		stop_pulling()
		
```

Personally i would add this a chance.
With the amount of bullets being hit at you from a marine firestorm its likely going to be triggered anyway, but a single marine isn't going to stop anything.

Specific value can be tweaked.

Reply:
Is this a 5% or 50% chance to drop?

The idea is that any source of damage stops dragging (FIRE, bullets, laser guns etc.)

Reply:
This would be 5%.

Persoanally don't agree that every instance of damage should cause you to stop dragging, but some threshold (or in this case, random chance).

Reply:
Note that this isn't really a flat 5% since there are a number of things that apply multiple instances of damage (shotguns, fire etc).

Reply:
While multiple instances of damage could make it cumbersome to use, I think the percentage should be slightly higher as there are already castes that can quite easily transport items that can be dragged i.e wraith. This just gives a bit of counterplay from the marine's side outside of needing to have a stun weapon. Would 30 or 25% be better?